src/config/datadog.rs

```diff
@@ -0,0 +1,143 @@
+use super::{ComponentId, Config, SinkOuter, SourceOuter};
+use crate::{
+    sinks::datadog::metrics::DatadogConfig, sources::internal_metrics::InternalMetricsConfig,
+};
+use serde::{Deserialize, Serialize};
+
+static INTERNAL_METRICS_KEY: &str = "#datadog_internal_metrics";
+static DATADOG_METRICS_KEY: &str = "#datadog_metrics";
+
+/// Error type for failed attempts to attach Datadog observability to a `Config`.
+pub enum AttachError {
+    NotEnabled,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+#[serde(default, deny_unknown_fields)]
+pub struct Options {
+    #[serde(default = "default_enabled")]
+    pub enabled: bool,
+
+    #[serde(default = "default_api_key")]
+    pub api_key: Option<String>,
+
+    #[serde(default = "default_reporting_interval_secs")]
+    pub reporting_interval_secs: u64,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Self {
+            enabled: default_enabled(),
+            api_key: default_api_key(),
+            reporting_interval_secs: default_reporting_interval_secs(),
+        }
+    }
+}
+
+/// By default, the Datadog feature is enabled.
+fn default_enabled() -> bool {
+    true
+}
+
+/// By default, no API key is provided.
+fn default_api_key() -> Option<String> {
+    None
+}
```

Comment:
Note that `None` is the normal default for any `Option`, so this is technically not needed.

Reply:
Updated.